Sourcery/Parsing/Parser.swift

```diff
@@ -523,18 +553,86 @@ extension Parser {
 
         return insideBlock ? .begin(annotations) : .annotations(annotations)
     }
+    
+    fileprivate func parseTypealiases(from source: [String: SourceKitRepresentable], containingType: Type?, processed: [[String: SourceKitRepresentable]]) -> [Typealias] {
+        var contentToParse = self.contents
+        
+        //replace all processed substructures with whitespaces so that we don't process their typealiases again
+        for substructure in processed {
+            if let range = SubstringIdentifier.key.range(for: substructure)
+                .flatMap({ contentToParse.byteRangeToNSRange(start: Int($0.offset), length: Int($0.length)) }) {
+            
+                let replacement = String(repeating: " ", count: range.length)
+                contentToParse = contentToParse.bridge().replacingCharacters(in: range, with: replacement)
+            }
+        }
+        
+        if containingType != nil {
+            if let body = extract(.body, from: source, contents: contentToParse) {
+                return parseTypealiases(SyntaxMap(file: File(contents: body)).tokens, contents: body)
+            } else {
+                return []
+            }
+        } else {
+            return parseTypealiases(SyntaxMap(file: File(contents: contentToParse)).tokens, contents: contentToParse)
+        }
+    }
+    
+    private func parseTypealiases(_ tokens: [SyntaxToken], contents: String, existingTypealiases: [Typealias] = []) -> [Typealias] {
+        var typealiases = existingTypealiases
+        
+        for (index, token) in tokens.enumerated() {
+            if token.type == "source.lang.swift.syntaxtype.keyword",
+                extract(token, contents: contents) == "typealias" {
+                
+                if index > 0,
+                    let accessLevel = extract(tokens[index - 1], contents: contents).flatMap(AccessLevel.init),
+                    accessLevel == .private || accessLevel == .fileprivate {
+                    continue
+                }
+                
+                guard let alias = extract(tokens[index + 1], contents: contents),
+                    let type = extract(tokens[index + 2], contents: contents) else {
+                        continue
+                }
+                
+                //get all subsequent type identifiers
+                var subtypes = [type]
```

Comment:
I refactored some code and have changes as I'm adding support for one typealias to reference another, but I'm hitting issue with this method(didn't touch it), this test should pass right?:
```swift
fit("extracts typealiases of other typealiases") {
    expect(sut?.parseContents("typealias Foo = Int; typealias Bar = Foo").typealiases)
        .to(contain([
            Typealias(aliasName: "Foo", typeName: "Int"),
            Typealias(aliasName: "Bar", typeName: "Foo"),
            ]))
}
```